Sources/Utility/ArgumentParser.swift

```diff
@@ -373,8 +373,16 @@ public final class ArgumentParser: ArgumentParserProtocol {
     /// The current argument being parsed.
     private var currentArgument: String!
 
+    /// The current associated value in a `--foo=bar` style argument.
+    private var associatedArgumentValue: String?
+
     /// Provides next argument from iterator or throws.
     public func next() throws -> String {
```

Comment:
Should this really be part of `next()` versus a more explicit function for requesting the "next value"? If this is all `next()` is for, maybe it should be `getNextValue()` or something?

The problem with `next()` is its definition as part of the iterator protocol, which conflicts with how it is being used in this patch a little bit.